build/changelog/ChangelogBuilder.php

```diff
@@ -0,0 +1,149 @@
+<?php
+namespace Aws\Build\Changelog;
+
+/**
+ * @internal
+ */
+
+class ChangelogBuilder 
+{
+  
+  public $newServiceFlag = False;
+
+  public function readChangelog($dir)
+  {
+    $dir = $dir."nextrelease/";
+    $changelogEntries = [];
+    if ( !is_dir($dir) || !$dh = opendir($dir) ) {
+        throw new \Exception("nextrelease directory doesn't exists or is not readable at location $dir");
+    } else {
+      //Ignore any files starting with a (.) dot 
+      $files=preg_grep('/^([^.])/', scandir($dir));
+      if ( empty($files) ) {
```

Comment:
Sorry perhaps my example mislead you there,  I'm actually suggesting `if (empty($files)) {`